mm2src/coins/lightning/ln_events.rs

```diff
@@ -222,28 +299,114 @@ impl LightningEventHandler {
             "Handling PaymentSent event for payment_hash: {}",
             hex::encode(payment_hash.0)
         );
-        if let Some(payment) = self.outbound_payments.lock().get_mut(&payment_hash) {
-            payment.preimage = Some(payment_preimage);
-            payment.status = HTLCStatus::Succeeded;
-            payment.fee_paid_msat = fee_paid_msat;
-            log::info!(
-                "Successfully sent payment of {} millisatoshis with payment hash {}",
-                payment.amt_msat.unwrap_or_default(),
-                hex::encode(payment_hash.0)
-            );
+        let persister = self.persister.clone();
+        spawn(async move {
+            if let Ok(Some(mut payment_info)) = persister
+                .get_payment_from_sql(payment_hash)
+                .await
+                .error_log_passthrough()
+            {
+                payment_info.preimage = Some(payment_preimage);
+                payment_info.status = HTLCStatus::Succeeded;
+                payment_info.fee_paid_msat = fee_paid_msat;
+                payment_info.last_updated = now_ms() / 1000;
+                let amt_msat = payment_info.amt_msat;
+                if let Err(e) = persister
+                    .add_or_update_payment_in_sql(payment_info)
+                    .await
+                    .error_log_passthrough()
+                {
+                    log::error!("{}", e);
+                }
```

Comment:
I don't understand this part.

According to log_passthrough:

```rust
 #[track_caller]
    fn error_log_passthrough(self) -> Self {
        if let Err(e) = &self {
            let location = std::panic::Location::caller();
            let file = gstuff::filename(location.file());
            let line = location.line();
            error!("{}:{}] {}", file, line, e);
        }
        self
    }
```
    
    Is the log written to console twice?

Reply:
Must have kept `error_log_passthrough()` from before adding `if let Err(e)`, will fix.